### PR TITLE
feat: allow the use of custom units to format the duration

### PR DESCRIPTION
**Pull request by moisespsena**

### CODE REVIEW

File: unit.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+// 		- must singular units: `"year,week,day,hour,minute,second,millisecond,microsecond"`
+// 		- mixed units: `"year,week:weeks,day,hour,minute:minutes,second,millisecond,microsecond"`
+// 		- singular and plural pair units: `"year:wers,week:weeks,day:days,hour:hours,minute:minutes,second:seconds,millisecond:millliseconds,microsecond:microsseconds"`
+func (coder UnitsCoder) Decode(s string) (units Units, err error) {
+	parts := strings.Split(s, coder.UnitsSep)
+	if len(parts) != 8 {
+		err = fmt.Errorf("bad parts length")
+		return
+	}
+	var ru = reflect.ValueOf(&units).Elem()
```
Comment (hako) [inline unit.go:68]: Is there any reason why we are using reflect here?

File: unit.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package durafmt
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// DefaultUnitsCoder default unots coder using `":"` as PluralSep and `","` as UnitsSep
```
Comment (hako) [inline unit.go:9]: typo in 'unots'

File: unit.go (hunk 1 above)
Comment (moisespsena) [reply unit.go:68]: just for ease of use. On second thought, for performance reasons, it is better not to use it. I'll change that. Thanks for mentioning.

File: unit.go (hunk 2 above)
Comment (moisespsena) [reply unit.go:9]: no, 'units'. It went wrong.

File: unit.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+//	Second, Millisecond and Microsecond units) separated by `UnitsSep` char
+// 	- Examples with `UnitsCoder{PluralSep: ":", UnitsSep = ","}`
+// 		- must singular units: `"year,week,day,hour,minute,second,millisecond,microsecond"`
+// 		- mixed units: `"year,week:weeks,day,hour,minute:minutes,second,millisecond,microsecond"`
+// 		- singular and plural pair units: `"year:wers,week:weeks,day:days,hour:hours,minute:minutes,second:seconds,millisecond:millliseconds,microsecond:microsseconds"`
+func (coder UnitsCoder) Decode(s string) (units Units, err error) {
+	parts := strings.Split(s, coder.UnitsSep)
+	if len(parts) != 8 {
+		err = fmt.Errorf("bad parts length")
+		return
```
Comment (hako) [inline unit.go:65]: I think we should be explicit with the return values.

File: unit.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+			return false
+		}
+		return true
+	}
+
+	if !parse("Year", parts[0], &units.Year) {
+		return
+	}
+	if !parse("Week", parts[1], &units.Week) {
+		return
```
Comment (hako) [inline unit.go:87]: Same with the above, with explicit with the return values, would it be possible to cover (from Week to Microsecond) with tests? [See coverage report for more info.](https://codecov.io/gh/hako/durafmt/commit/5c8cfd540029d4f45569727cd35f1ce2d7bb3295)

**Pull request merged**